authz/client/src/main/java/org/keycloak/authorization/client/util/HttpMethod.java

```diff
@@ -92,7 +92,7 @@ public R execute(HttpResponseProcessor<R> responseProcessor) {
             int statusCode = statusLine.getStatusCode();
 
             if (statusCode < 200 || statusCode >= 300) {
-                throw new HttpResponseException("Unexpected response from server: " + statusCode + " / " + statusLine.getReasonPhrase(), statusCode, statusLine.getReasonPhrase(), bytes);
+                throw new HttpResponseException("Unexpected response from server: " + statusCode + " / " + statusLine.getReasonPhrase()  + " Response : " + new String(bytes), statusCode, statusLine.getReasonPhrase(), bytes);
```

Comment:
For debugging purposes, we can use Java Logging and log the response accordingly when level `FINE` is set.